From 475fd5466d6e186a42c54060c45c2b90b07f0eec Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:07 -0400
Subject: [usb] option: add device ID for SpeedUp SU9800 usb 3g modem

Message-id: <1410887463-20674-536-git-send-email-dzickus@redhat.com>
Patchwork-id: 95625
O-Subject: [RHEL7 PATCH 535/591] USB: option: add device ID for SpeedUp SU9800 usb 3g modem
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 1cab4c68e339086cdaff7535848e878e8f261fca
Author: Oliver Neukum <oneukum@suse.de>
Date:   Tue May 20 16:27:40 2014 +0200

    USB: option: add device ID for SpeedUp SU9800 usb 3g modem

    Reported by Alif Mubarak Ahmad:

    This device vendor and product id is 1c9e:9800
    It is working as serial interface with generic usbserial driver.
    I thought it is more suitable to use usbserial option driver, which has
    better capability distinguishing between modem serial interface and
    micro sd storage interface.

    [ johan: style changes ]

    Signed-off-by: Oliver Neukum <oneukum@suse.de>
    Tested-by: Alif Mubarak Ahmad <alive4ever@live.com>
    Cc: <stable@vger.kernel.org>
    Signed-off-by: Johan Hovold <johan@kernel.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 59c3108..f4fd59c 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -352,6 +352,9 @@ static void option_instat_callback(struct urb *urb);
 /* Zoom */
 #define ZOOM_PRODUCT_4597   0x9607
 
+/* SpeedUp SU9800 usb 3g modem */
+#define SPEEDUP_PRODUCT_SU9800   0x9800
+
 /* Haier products */
 #define HAIER_VENDOR_ID    0x201e
 #define HAIER_PRODUCT_CE100   0x2009
@@ -1577,6 +1580,7 @@ static const struct usb_device_id option_ids[] = {
  { USB_DEVICE(LONGCHEER_VENDOR_ID, FOUR_G_SYSTEMS_PRODUCT_W14),
      .driver_info = (kernel_ulong_t)&four_g_w14_blacklist
    },
+ { USB_DEVICE_INTERFACE_CLASS(LONGCHEER_VENDOR_ID, SPEEDUP_PRODUCT_SU9800, 0xff) },
  { USB_DEVICE(LONGCHEER_VENDOR_ID, ZOOM_PRODUCT_4597) },
  { USB_DEVICE(LONGCHEER_VENDOR_ID, IBALL_3_5G_CONNECT) },
  { USB_DEVICE(HAIER_VENDOR_ID, HAIER_PRODUCT_CE100) },
-- 
1.7.1